src/NServiceBus.AcceptanceTests/Diagnostics/When_ambient_trace_in_message_session.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace NServiceBus.AcceptanceTests.Diagnostics
+{
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    [NonParallelizable] // Ensure only activities for the current test are captured
```

Comment:
we could create a base class that provides the attribute and also does the trace subscription?